clients/drcachesim/reader/v2p_reader.cpp

```diff
@@ -33,83 +33,114 @@
-                       elem.c_str());
-                return false;
-            }
-            virtual_address = std::stoull(elem, nullptr, 0);
+            virtual_address =
+                static_cast<addr_t>(std::stoull(key_val_pair[1], nullptr, 0));
             continue;
         }
-        found = line.rfind(physical_address_key);
+
+        found = line.find(physical_address_key);
         if (found != std::string::npos) {
+            std::vector<std::string> key_val_pair = split_by(line, separator);
+            if (key_val_pair.size() != 2) {
+                error_ss << "ERROR: " << physical_address_key
+                         << " key or value not found.";
+                return error_ss.str();
+            }
             if (virtual_address == 0) {
-                ERRMSG("ERROR: no corresponding virtual_address for this "
-                       "physical_address.\n");
-                return false;
+                error_ss << "ERROR: no corresponding " << virtual_address_key
+                         << " for this " << physical_address_key << " " << key_val_pair[1]
+                         << ".";
+                return error_ss.str();
             }
-            std::stringstream ss(line);
-            std::string elem;
-            if (!std::getline(ss, elem, ':')) {
-                ERRMSG("ERROR: physical_address key not found. Found %s instead.\n",
-                       elem.c_str());
-                return false;
+            addr_t physical_address =
+                static_cast<addr_t>(std::stoull(key_val_pair[1], nullptr, 0));
+            v2p_info.v2p_map[virtual_address] = physical_address;
+        }
+        virtual_address = 0;
+
+        found = line.find(page_size_key);
+        if (found != std::string::npos) {
+            std::vector<std::string> key_val_pair = split_by(line, separator);
+            if (key_val_pair.size() != 2) {
+                error_ss << "ERROR: " << page_size_key << " key or value not found.";
+                return error_ss.str();
             }
-            if (!std::getline(ss, elem, ':')) {
-                ERRMSG("ERROR: physical_address value not found. Found %s instead.\n",
-                       elem.c_str());
-                return false;
+            v2p_info.page_size =
+                static_cast<size_t>(std::stoull(key_val_pair[1], nullptr, 0));
+            continue;
+        }
+
+        found = line.find(page_count_key);
+        if (found != std::string::npos) {
+            std::vector<std::string> key_val_pair = split_by(line, separator);
+            if (key_val_pair.size() != 2) {
+                error_ss << "ERROR: " << page_count_key << " key or value not found.";
+                return error_ss.str();
             }
-            uint64_t physical_address = std::stoull(elem, nullptr, 0);
-            v2p_map[virtual_address] = physical_address;
+            v2p_info.page_count = std::stoull(key_val_pair[1], nullptr, 0);
```

Comment:
Is it an error for this to appear twice?  Ditto for page size and bytes mapped.

Reply:
It's not an error, page size and bytes mapped do appear twice in the v2p.textproto files we generate.
Here we're just taking the last value for these fields, but it's probably a good idea to get the first one and check if the following ones match (or return and error otherwise).
Done.